main.go

```diff
@@ -241,12 +241,18 @@ func main() {
 	templatesCleaned := make(chan struct{})
 	go constrainttemplate.TearDownState(cli, templatesCleaned)
 
+	// Make sure certs are generated and valid.
+	certsValid := make(chan struct{})
+	go webhook.EnsureValidCerts(cli, certsValid)
```

Comment:
We have a flag that allows the user to disable cert generation. Currently it is enforced as part of adding the webhook:

https://github.com/open-policy-agent/gatekeeper/blob/bba52f962787e005efe3ca93ad2d72d7c7eb3ceb/pkg/webhook/policy.go#L70-L77

The main reason why we enforce it as part of adding the webhook is that cert generation is only necessary when the webhook is in use, so if the webhook is disabled, cert generation is moot.

I'm not 100% sure what the correct solution is here. We could pass the channel to `webhoiok.AddToManager()`, which could just close the channel if cert generation is disabled.

We could decouple the cert generation completely and do what you had suggested, which was to defer adding anything else to the manager until certs are successfully written.

Maybe some other pattern that's not coming to mind. LMK if there is anything I can do to help here.

Reply:
How about checking the `disableCertRotation` flag to condition on this code block?

Reply:
+1 to the general notion of checking the flag. Doing it directly is similar to the "decouple completely", at which point, I would like to start the cert rotator before the manager, but it'd also be cool if the webhook module could handle its own flags so that we have code isolation. That may not be possible.

Reply:
Thanks all for the review!

I added checking the flag into the new PR #581  , where the manager is started only with cert controllers added. Then once the certs are generated, the other controllers and the webhooks are added on the fly.

PTAL and review the new PR. I'm going to close this one for now. Thanks!